### PR TITLE
Photo implementation

### PR DESCRIPTION
**Pull request by IvanTyulyandin**
Add camera imitation.
V4l2: choose existing pixel format and after taking shot convert image to RGB888, early converting to RGB888 was in trikHal.

### CODE REVIEW

File: trikControl/src/brick.cpp [hunk 1]
```diff
@@ -550,9 +550,9 @@ void Brick::createDevice(const QString &port)
 			connect(mSoundSensors[port], SIGNAL(stopped()), this, SIGNAL(stopped()));
 		} else if (deviceClass == "fifo") {
 			mFifos.insert(port, new Fifo(port, mConfigurer, *mHardwareAbstraction));
-        } else if (deviceClass == "camera") {
-			QScopedPointer<CameraDeviceInterface> tmp (new CameraDevice(mMediaPath, mConfigurer, *mHardwareAbstraction));
-			mCamera.swap(tmp);
+		} else if (deviceClass == "camera") {
+					QScopedPointer<CameraDeviceInterface> tmp (new CameraDevice(mMediaPath, mConfigurer, *mHardwareAbstraction));
```
Comment (ZiminGrigory) [inline trikControl/src/brick.cpp]: Слишком много табов.

File: trikControl/src/imitationCameraImplementation.cpp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+using namespace trikControl;
+
+ImitationCameraImplementation::ImitationCameraImplementation(const QStringList &filter, const QString &path)
+	: filters(filter)
+{
+	setTempDir(path);
+	filesList = QDir(getTempDir()).entryInfoList(filters);
+	if (filesList.isEmpty()) {
+		QLOG_ERROR() << "No files in directory " << getTempDir()
+					 << " matching regexps " << filters;
```
Comment (ZiminGrigory) [inline trikControl/src/imitationCameraImplementation.cpp]: Стайлгайд советует 2 таба.

File: trikControl/src/imitationCameraImplementation.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	if (filesList.size() > 1) {
+		QLOG_INFO() << "Few files matching regexps " << filters
+					<< " in file " << getTempDir();
```
Comment (ZiminGrigory) [inline trikControl/src/imitationCameraImplementation.cpp]: и тут

File: trikControl/src/imitationCameraImplementation.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+		auto f = filesList[++cur%=filesList.size()];
+		QImage imgOrig(f.fileName());
+
+		if (! imgOrig.isNull()) {
+			QLOG_INFO() << "Opening file " << f.fileName();
+		} else {
+			QLOG_ERROR() << "Can not open file " << f.fileName();
+			return QVector<uint8_t>();
+		}
+		return qImageToQVector(imgOrig);
```
Comment (ZiminGrigory) [inline trikControl/src/imitationCameraImplementation.cpp]: пустую строку

**Comment by iakov** [comment]
/home/travis/build/trikset/trikRuntime/trikControl/include/trikControl/brickInterface.h:117: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/include/trikControl/brickInterface.h:118: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/include/trikControl/cameraDeviceInterface.h:1: error: No copyright notice
/home/travis/build/trikset/trikRuntime/trikControl/include/trikControl/cameraDeviceInterface.h:10: error: Classes must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikControl/include/trikControl/cameraDeviceInterface.h:14: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/include/trikControl/cameraDeviceInterface.h:14: error: Methods must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikControl/include/trikControl/cameraDeviceInterface.h:16: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/include/trikControl/cameraDeviceInterface.h:16: error: Methods must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikControl/src/brick.cpp:398: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/src/brick.cpp:399: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/src/brick.cpp:400: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/src/brick.cpp:401: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/src/brick.cpp:553: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/src/brick.cpp:554: error: line is longer than 120 characters
/home/travis/build/trikset/trikRuntime/trikControl/src/brick.cpp:556: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/src/brick.h:121: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/src/brick.h:172: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/src/cameraDevice.cpp:1: error: No copyright notice
/home/travis/build/trikset/trikRuntime/trikControl/src/cameraDevice.cpp:40: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/src/cameraDevice.h:1: error: No copyright notice
/home/travis/build/trikset/trikRuntime/trikControl/src/cameraDevice.h:20: error: Classes must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikControl/src/cameraImplementationInterface.h:1: error: No copyright notice
/home/travis/build/trikset/trikRuntime/trikControl/src/cameraImplementationInterface.h:9: error: Classes must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikControl/src/cameraImplementationInterface.h:12: error: Methods must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikControl/src/cameraImplementationInterface.h:16: error: Methods must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikControl/src/cameraImplementationInterface.h:18: error: Methods must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikControl/src/qtCameraImplementation.cpp:1: error: No copyright notice
/home/travis/build/trikset/trikRuntime/trikControl/src/qtCameraImplementation.cpp:25: error: line is longer than 120 characters
/home/travis/build/trikset/trikRuntime/trikControl/src/qtCameraImplementation.cpp:42: error: line is longer than 120 characters
/home/travis/build/trikset/trikRuntime/trikControl/src/qtCameraImplementation.cpp:54: error: line is longer than 120 characters
/home/travis/build/trikset/trikRuntime/trikControl/src/qtCameraImplementation.cpp:68: error: line is longer than 120 characters
/home/travis/build/trikset/trikRuntime/trikControl/src/qtCameraImplementation.cpp:82: error: line is longer than 120 characters
/home/travis/build/trikset/trikRuntime/trikControl/src/qtCameraImplementation.h:1: error: No copyright notice
/home/travis/build/trikset/trikRuntime/trikControl/src/qtCameraImplementation.h:11: error: Classes must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikControl/src/qtCameraImplementation.h:15: error: Methods must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikControl/src/qtCameraImplementation.h:19: error: String starts with spaces
/home/travis/build/trikset/trikRuntime/trikControl/src/v4l2CameraImplementation.cpp:1: error: No copyright notice
/home/travis/build/trikset/trikRuntime/trikControl/src/v4l2CameraImplementation.cpp:6: error: line is longer than 120 characters
/home/travis/build/trikset/trikRuntime/trikControl/src/v4l2CameraImplementation.h:1: error: No copyright notice
/home/travis/build/trikset/trikRuntime/trikControl/src/v4l2CameraImplementation.h:14: error: Classes must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikControl/src/v4l2CameraImplementation.h:17: error: Methods must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikControl/src/v4l2CameraImplementation.h:19: error: Methods must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikGui/trikGuiApplication.cpp:43: error: Comment text shall be separated by one space from comment symbol
/home/travis/build/trikset/trikRuntime/trikGui/trikGuiApplication.cpp:46: error: Comment text shall be separated by one space from comment symbol
/home/travis/build/trikset/trikRuntime/trikHal/src/trik/trikV4l2VideoDevice.cpp:1: error: No copyright notice
/home/travis/build/trikset/trikRuntime/trikHal/src/trik/trikV4l2VideoDevice.cpp:64: error: line is longer than 120 characters
/home/travis/build/trikset/trikRuntime/trikHal/src/trik/trikV4l2VideoDevice.cpp:65: error: line is longer than 120 characters
/home/travis/build/trikset/trikRuntime/trikHal/src/trik/trikV4l2VideoDevice.cpp:188: error: NULL is forbidden, use nullptr instead
/home/travis/build/trikset/trikRuntime/trikHal/src/trik/trikV4l2VideoDevice.h:1: error: No copyright notice
/home/travis/build/trikset/trikRuntime/trikHal/src/trik/trikV4l2VideoDevice.h:9: error: Classes must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikHal/src/trik/trikV4l2VideoDevice.h:13: error: Methods must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikHal/src/trik/trikV4l2VideoDevice.h:15: error: Methods must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikHal/src/trik/trikV4l2VideoDevice.h:16: error: Methods must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikHal/src/trik/trikV4l2VideoDevice.h:19: error: Methods must have comments in doxygen style (starting with ///)
/home/travis/build/trikset/trikRuntime/trikHal/src/trik/trikV4l2VideoDevice.h:21: error: Methods must have comments in doxygen style (starting with ///)

**Pull request merged**